src/Generator/Check24DE.php

```diff
@@ -38,127 +65,264 @@ public function __construct(ArrayHelper $arrayHelper)
-                    'name' 				=> $this->elasticExportHelper->getName($item, $settings) . (strlen($variationName) ? ' ' . $variationName : ''),
-                    'description' 		=> $this->elasticExportHelper->getDescription($item, $settings),
-                    'category_path' 	=> $this->elasticExportHelper->getCategory((int)$item['data']['defaultCategories'][0]['id'], $settings->get('lang'), $settings->get('plentyId')),
-                    'price' 			=> number_format((float)$this->idlVariations[$item['id']]['variationRetailPrice.price'], 2, '.', ''),
-                    'price_per_unit'	=> $this->elasticExportHelper->getBasePrice($item, $this->idlVariations),
-                    'link' 				=> $this->elasticExportHelper->getUrl($item, $settings, true, false),
-                    'image_url'			=> $this->elasticExportHelper->getMainImage($item, $settings),
-                    'delivery_time' 	=> $this->elasticExportHelper->getAvailability($item, $settings, false),
-                    'delivery_cost' 	=> $shippingCost,
-                    'pzn' 				=> '',
-                    'stock' 			=> $this->idlVariations[$item['id']]['variationStock.stockNet'],
-                    'weight' 			=> $item['data']['variation']['weightG'],
-                ];
-
-                $this->addCSVContent(array_values($data));
-            }
+            } while ($elasticSearch->hasNext());
         }
+
+        $this->getLogger(__METHOD__)->debug('ElasticExportCheck24DE::logs.fileGenerationDuration', [
+            'Whole file generation duration' => microtime(true) - $startTime,
+        ]);
+    }
+
+    /**
+     * Creates the header of the CSV file.
+     *
+     * @return array
+     */
+    private function head():array
+    {
+        return array(
+            'id',
+            'manufacturer',
+            'mpnr',
+            'ean',
+            'name',
+            'description',
+            'category_path',
+            'price',
+            'price_per_unit',
+            'link',
+            'image_url',
+            'delivery_time',
+            'delivery_cost',
+            'pzn',
+            'stock',
+            'weight'
+        );
     }
 
     /**
-     * Get the SKU.
-     * @param  array $item
+     * Creates the variation row and prints it into the CSV file.
+     *
+     * @param $variation
+     * @param KeyValue $settings
+     */
+    private function buildRow($variation, KeyValue $settings)
+    {
+        // get the price
+        $priceList = $this->elasticExportPriceHelper->getPriceList($variation, $settings);
+
+        // only variations with the Retail Price greater than zero will be handled
+        if(!is_null($priceList['price']) && $priceList['price'] > 0)
+        {
+            $variationName = $this->elasticExportHelper->getAttributeValueSetShortFrontendName($variation, $settings);
+
+            $shippingCost = $this->getShippingCost($variation);
```

Comment:
You don't need to put that into a variable, if you wanted just to push the result of the function to the feed. If you would working with the value and something depends on the value, then it would make sense.